frontend/src/metabase-types/analytics/account.ts

```diff
@@ -0,0 +1,20 @@
+type AccountEventSchema = {
+  event: string;
+  version?: string | null;
+  link?: string | null;
+};
+
+type ValidateEvent<
+  T extends AccountEventSchema &
+    Record<Exclude<keyof T, keyof AccountEventSchema>, never>,
+> = T;
+
+export type NewUserCreatedEvent = ValidateEvent<{
```

Comment:
I got really confused because we use `ValidateEvent` also in the simple event file, can we somehow differentiate them?

Maybe we can name those `ValidateAccountEvent` to be specific to the schema they use, and then the generic one could be `ValidateSimpleEvent`

Reply:
all `ValidateEvent` helpers are not expected and schema-specific. Is this still confusing?

Reply:
I'm just afraid we on big files we may not see if they are imported from somewhere else or local. Can we at least rename the one for the simple event to `ValidateSimpleEvent`?
In that way we'd be able to tell at first sight what kind of event they are

Reply:
they are **all** local. Does your comment still apply?

Reply:
If we're going to have all SimpleEvents on that file then I guess we're ok.
But we usually have a `analytics.ts` file close to where we use it and we have local trackXXX functions (and types) there

Reply:
But I guess for now we can go like this and we'll see how many events we'll have

Reply:
@npretto you'd need to define event types that 100% match things that go to snowplow in `analytics` folder. There can be local types in features that unrelated to schemas.

Reply:
I'll need to create a new "SimpleEvent" tomorrow so I'll ask both you and @Somtom for a review to make sure I'm doing it right, we could then use that to show how easy it is to use it